code/modules/religion/sparring/sparring_datum.dm

```diff
@@ -0,0 +1,293 @@
+	unhook_signals(opponent)
+	chaplain.remove_filter("sparring_outline")
+	opponent.remove_filter("sparring_outline")
+
+/datum/sparring_match/proc/end_match(mob/living/carbon/human/winner, mob/living/carbon/human/loser, violation_victory = FALSE)
+	cleanup_sparring_match()
+	to_chat(chaplain, span_boldannounce("[violation_victory ? "[loser] DISQUALIFIED!" : ""]  [winner] HAS WON!"))
+	to_chat(opponent, span_boldannounce("[violation_victory ? "[loser] DISQUALIFIED!" : ""]  [winner] HAS WON!"))
+	win(winner, loser, violation_victory)
+	lose(loser, winner)
+	if(stakes_condition != YOUR_SOUL)
+		var/healing_message = "You may want to heal up the loser now."
+		if(winner == chaplain)
+			healing_message += " Your bible will heal the loser for awhile."
+		to_chat(winner, span_notice(healing_message))
+	qdel(src)
+
+///most of the effects are handled on `lose()` instead.
+/datum/sparring_match/proc/win(mob/living/carbon/human/winner, mob/living/carbon/human/loser, violation_victory)
+	switch(stakes_condition)
+		if(HOLY_MATCH)
+			if(winner == chaplain)
+				if(violation_victory)
+					to_chat(winner, span_warning("[GLOB.deity] is not entertained from a matched decided by violations. No favor awarded..."))
+				else
+					to_chat(winner, span_nicegreen("You've won favor with [GLOB.deity]!"))
+				var/datum/religion_sect/spar/sect = GLOB.religious_sect
+				sect.adjust_favor(1, winner)
+				sect.past_opponents += WEAKREF(loser)
+		if(MONEY_MATCH)
+			to_chat(winner, span_nicegreen("You've won all of [loser]'s money!"))
+		if(YOUR_SOUL)
+			to_chat(winner, span_nicegreen("You've won [loser]'s SOUL!"))
+
+/datum/sparring_match/proc/lose(mob/living/carbon/human/loser, mob/living/carbon/human/winner)
+	switch(stakes_condition)
+		if(HOLY_MATCH)
+			if(loser == chaplain)
+				var/datum/religion_sect/spar/sect = GLOB.religious_sect
+				sect.matches_lost++
+				if(sect.matches_lost < 3)
+					to_chat(loser, span_userdanger("[GLOB.deity] is angry you lost in their name!"))
+					return
+				to_chat(loser, span_userdanger("[GLOB.deity] is enraged by your lackluster sparring record!"))
+				lightningbolt(loser)
+				SEND_SIGNAL(loser, COMSIG_ADD_MOOD_EVENT, "sparring", /datum/mood_event/banished)
+				loser.mind.holy_role = NONE
+				to_chat(loser, span_userdanger("You have been excommunicated! You are no longer holy!"))
+		if(MONEY_MATCH)
+			to_chat(loser, span_userdanger("You've lost all your money to [winner]!"))
+			var/datum/bank_account/loser_account = loser.get_bank_account()
+			var/datum/bank_account/winner_account = winner.get_bank_account()
+			if(!loser_account || !winner_account)//the winner is pretty owned in this case but whatever shoulda read the fine print of the contract
+				return
+			winner_account.transfer_money(loser_account, loser_account.account_balance)
+		if(YOUR_SOUL)
+			to_chat(loser, span_userdanger("You've lost ownership over your soul to [winner]!"))
+			var/obj/item/soulstone/anybody/sparring/shard = new(get_turf(loser))
```

Comment:
make sure that the loser actually still exists and has a turf to put the soulstone on